include/core/Godot.hpp

```diff
@@ -43,10 +43,14 @@ public:
 		script->set_library(godot::get_wrapper<godot::GDNativeLibrary>((godot_object *)godot::gdnlib));                                             \
 		script->set_class_name(#Name);                                                                                                              \
 		Name *instance = godot::as<Name>(script->new_());                                                                                           \
+		godot::Reference *ref = (godot::Reference *)instance;                                                                                       \
```

Comment:
Should not use C-style cast. This will "always succeed", even when `instance` isn't a reference, resulting in undefined behavior. Use `Object::cast_to` instead.